ffi/capi/dart/package/lib/lib.dart

```diff
@@ -0,0 +1,11466 @@
+  void ceil(int position) {
+    _ceilFfi(_underlying, position);
+  }
+
+  static late final _ceilFfi = _capi<
+          ffi.NativeFunction<
+              ffi.Void Function(ffi.Pointer<ffi.Opaque>,
+                  ffi.Int16)>>('ICU4XFixedDecimal_ceil')
+      .asFunction<void Function(ffi.Pointer<ffi.Opaque>, int)>(isLeaf: true);
+
+  /// See the [Rust documentation for `half_ceil`](https://docs.rs/fixed_decimal/latest/fixed_decimal/struct.FixedDecimal.html#method.half_ceil) for more information.
+  void halfCeil(int position) {
+    _halfCeilFfi(_underlying, position);
+  }
+
+  static late final _halfCeilFfi = _capi<
+          ffi.NativeFunction<
+              ffi.Void Function(ffi.Pointer<ffi.Opaque>,
+                  ffi.Int16)>>('ICU4XFixedDecimal_half_ceil')
+      .asFunction<void Function(ffi.Pointer<ffi.Opaque>, int)>(isLeaf: true);
+
+  /// See the [Rust documentation for `floor`](https://docs.rs/fixed_decimal/latest/fixed_decimal/struct.FixedDecimal.html#method.floor) for more information.
+  void floor(int position) {
+    _floorFfi(_underlying, position);
+  }
+
+  static late final _floorFfi = _capi<
+          ffi.NativeFunction<
+              ffi.Void Function(ffi.Pointer<ffi.Opaque>,
+                  ffi.Int16)>>('ICU4XFixedDecimal_floor')
+      .asFunction<void Function(ffi.Pointer<ffi.Opaque>, int)>(isLeaf: true);
+
+  /// See the [Rust documentation for `half_floor`](https://docs.rs/fixed_decimal/latest/fixed_decimal/struct.FixedDecimal.html#method.half_floor) for more information.
+  void halfFloor(int position) {
+    _halfFloorFfi(_underlying, position);
+  }
+
+  static late final _halfFloorFfi = _capi<
+          ffi.NativeFunction<
+              ffi.Void Function(ffi.Pointer<ffi.Opaque>,
+                  ffi.Int16)>>('ICU4XFixedDecimal_half_floor')
+      .asFunction<void Function(ffi.Pointer<ffi.Opaque>, int)>(isLeaf: true);
+
+  /// See the [Rust documentation for `half_even`](https://docs.rs/fixed_decimal/latest/fixed_decimal/struct.FixedDecimal.html#method.half_even) for more information.
+  void halfEven(int position) {
+    _halfEvenFfi(_underlying, position);
+  }
+
+  static late final _halfEvenFfi = _capi<
+          ffi.NativeFunction<
+              ffi.Void Function(ffi.Pointer<ffi.Opaque>,
+                  ffi.Int16)>>('ICU4XFixedDecimal_half_even')
+      .asFunction<void Function(ffi.Pointer<ffi.Opaque>, int)>(isLeaf: true);
+
+  /// Concatenates `other` to the end of `self`.
+  ///
+  /// If successful, `other` will be set to 0 and a successful status is returned.
+  ///
+  /// If not successful, `other` will be unchanged and an error is returned.
+  ///
+  /// See the [Rust documentation for `concatenate_end`](https://docs.rs/fixed_decimal/latest/fixed_decimal/struct.FixedDecimal.html#method.concatenate_end) for more information.
+  void concatenateEnd(ICU4XFixedDecimal other) {
+    final result = _concatenateEndFfi(_underlying,
```

Comment:
```suggestion
    throw UnimplementedError();
```

But this is probably WIP.

Reply:
So the problem here is that Dart doesn't have a typed list to represent a list of `ffi.Size`s, which are platform specific integer lengths. I'll just remove the `fromDart` and `asDart` for this.